runtime/Cpp/runtime/src/misc/IntervalSet.cpp

```diff
@@ -13,52 +13,31 @@
 using namespace antlr4;
 using namespace antlr4::misc;
 
-IntervalSet const IntervalSet::COMPLETE_CHAR_SET = []() {
-  IntervalSet complete = IntervalSet::of(Lexer::MIN_CHAR_VALUE, Lexer::MAX_CHAR_VALUE);
-  complete.setReadOnly(true);
-  return complete;
-}();
+IntervalSet const IntervalSet::COMPLETE_CHAR_SET = 
+    IntervalSet::of(Lexer::MIN_CHAR_VALUE, Lexer::MAX_CHAR_VALUE);
 
-IntervalSet const IntervalSet::EMPTY_SET = []() {
-  IntervalSet empty;
-  empty.setReadOnly(true);
-  return empty;
-}();
+IntervalSet const IntervalSet::EMPTY_SET;
 
-IntervalSet::IntervalSet() {
-  InitializeInstanceFields();
+IntervalSet::IntervalSet() : _intervals() {
 }
 
-IntervalSet::IntervalSet(const std::vector<Interval> &intervals) : IntervalSet() {
-  _intervals = intervals;
+IntervalSet::IntervalSet(const IntervalSet &set) : _intervals(set._intervals) {
 }
 
-IntervalSet::IntervalSet(const IntervalSet &set) : IntervalSet() {
-  addAll(set);
+IntervalSet::IntervalSet(IntervalSet&& set) : _intervals(std::move(set._intervals)) {
 }
 
-IntervalSet::IntervalSet(int n, ...) : IntervalSet() {
-  va_list vlist;
-  va_start(vlist, n);
-
-  for (int i = 0; i < n; i++) {
-    add(va_arg(vlist, int));
-  }
+IntervalSet::IntervalSet(std::vector<Interval>&& intervals) : _intervals(std::move(intervals)) {
```

Comment:
I replaced the varadic code with a templated version that is type safe and provides the same interface. There are two changes that are being mixed here because they're close by in the diff:

* The va_arg code was replaced with a templated version in the header file. The old va_arg code was removed. The interface is the same and any code using the old va_args based interface will now use the templated version (including type safety).
* At the same place in the file an internal constructor was placed to move in a vector of intervals.

They are distinct changes.